src/csv/lexer.cr

```diff
@@ -89,15 +89,15 @@ abstract class CSV::Lexer
       when '\0'
         raise "unclosed quote"
         break
-      when '"'
+      when @quote_char
         case next_char
-        when ','
+        when @separator
           check_last_empty_column
           break
         when '\r', '\n', '\0'
           break
-        when '"'
-          @buffer << '"'
+        when @quote_char
```

Comment:
I guess. One could always reopen the class and change the value. What's your specific concern?